test/nodetool/test_nodetool.py

```diff
@@ -0,0 +1,83 @@
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema", "-u", "us3r", "-pw", "secr3t")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password", "secr3t")
+    nodetool("compact", "system_schema", "-u", "us3r", "-pwf", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password-file", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "-pp")
+    nodetool("compact", "system_schema", "--print-port")
+
+
+def test_compact_all_keyspaces(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("GET", "/storage_service/keyspaces?type=all", json.dumps(["system", "system_schema"])),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact")
+
+
+def test_compact_keyspace(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema")
+
+
+def test_compact_table(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema?cf=columns")])
+
+    nodetool("compact", "system_schema", "columns")
+
+
+def test_compact_split_output_compatibility_argument(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema", "-s")
+    nodetool("compact", "system_schema", "--split-output")
+
+
+def test_compact_token_range_compatibility_argument(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema", "-st", "0", "-et", "1000")
+    nodetool("compact", "system_schema", "--start-token", "0", "--end-token", "1000")
```

Comment:
There's something I don't understand in these tests - shouldn't they end checking that all the expected requests were eaten up by the test?